targets/csharp/source/PlayFabSDK/source/Uunit/UUnitTestContext.cs

```diff
@@ -44,10 +44,97 @@ public UUnitTestContext(UUnitTestCase testInstance, Action<UUnitTestContext> tes
 
         public void EndTest(UUnitFinishState finishState, string resultMsg)
         {
-            EndTime = DateTime.UtcNow;
-            TestResultMsg = resultMsg;
-            FinishState = finishState;
-            ActiveState = UUnitActiveState.READY;
+            if (FinishState == UUnitFinishState.PENDING)
+            {
+                EndTime = DateTime.UtcNow;
+                TestResultMsg = resultMsg;
+                FinishState = finishState;
+                ActiveState = UUnitActiveState.READY;
+            }
+            else if (FinishState == UUnitFinishState.PASSED)
+            {
+                switch (finishState)
+                {
+                    case UUnitFinishState.PASSED:
+                        TestResultMsg += "Test try to Pass twice for some reason.\n";
+                        break;
+                    case UUnitFinishState.FAILED:
+                        TestResultMsg += "Test try to Fail after Passing.\n";
+                        break;
+                    case UUnitFinishState.SKIPPED:
+                        TestResultMsg += "Test try to be Skipped after Passing.\n";
+                        break;
+                    case UUnitFinishState.TIMEDOUT:
+                        TestResultMsg += "Test try to Timeout after Passing.\n";
+                        break;
+                    default:
+                        TestResultMsg += "How are you switching back to a Pending state from Passing.\n";
+                        break;
+                }
+            }
+            else if (FinishState == UUnitFinishState.FAILED)
+            {
+                switch (finishState)
+                {
+                    case UUnitFinishState.PASSED:
+                        TestResultMsg += "Test try to Pass after Failing.\n";
+                        break;
+                    case UUnitFinishState.FAILED:
+                        TestResultMsg += "Test try to Fail twice.\n";
```

Comment:
As I discussed earlier, we have declared that this test has failed twice. So we need to merge the old fail message and the new one, because the test-debugger will need to fix both issues.